File: 1400. Construct K Palindrome Strings/main.go
package main

func canConstruct(s string, k int) bool {
	if len(s) < k {
		return false
	}
	sCount := make(map[rune]int)
	for _, c := range s {
		sCount[c]++
	}
	oddCount := 0
	for _, v := range sCount {
		if v%2 > 0 {
			oddCount++
		}
	}
	return oddCount <= k
}

File: 1400. Construct K Palindrome Strings/main.py
from collections import Counter


class Solution:
    def canConstruct(self, s: str, k: int) -> bool:
        if len(s) < k:
            return False

        s_cnt = Counter(s)
        odd_count = 0
        for v in s_cnt.values():
            if v % 2:
                odd_count += 1
        
        return odd_count <= k

 